crates/ide/src/diagnostics/fixes.rs

```diff
@@ -102,6 +102,27 @@ impl DiagnosticWithFix for MissingOkInTailExpr {
     }
 }
 
+impl DiagnosticWithFix for RemoveThisSemicolon {
+    fn fix(&self, sema: &Semantics<RootDatabase>) -> Option<Fix> {
+        let root = sema.db.parse_or_expand(self.file)?;
+
+        let semicolon = self
+            .expr
```

Comment:
this would become just

```suggestion
            .expr.semicolon_token()?
```

if we store ExprStmt